File: Funciones/23. Tabla_9_for/js/tabla.js
/**
 * Tabla del 9 que me idique que dato es par e impar
 * Autor: Santiago GM
 * 03/04/2024
 */

let tabla;
let limite;

// como paramétros 

function tablas(ptablas,plimite){
    tabla = ptablas;
    limite = plimite;
    let contador;
    let resultado;
    let msg = "";

    for(contador=1; limite >= contador; contador ++ ){

        resultado = tabla * contador;

        msg += `${tabla} x ${contador} = ${resultado} `;
        msg+= validar(resultado);

    }
    return msg;
}

function validar(valida){
    let msg = "";

    if(valida%2==0){
        msg+= "El resultado es  par\n";
        
    }else{
        msg+="El resultado es impar\n"
    }
    return msg;
}

//Como Expresión

const tablasExp = function (ptablas,plimite){
    tabla = ptablas;
    limite = plimite
    let contador;
    let resultado;
    let msg = "";

    for(contador=1; limite >= contador; contador ++ ){

        resultado = tabla * contador;

        msg += `${tabla} x ${contador} = ${resultado}`;
        if(resultado%2==0){
            msg+= " Este es un par\n";
        }else{
            msg+=" Este es impar\n"
        }

    }
    return msg;
}

File: python/23. Tabla_9_for/utils/tabla.py
#Tabla del 9 que me idique que dato es par e impar
#Autor: Santiago GM


# como paramétros

def tablas(ptablas,plimite):
    tabla = ptablas
    limite = plimite
    msg = ""

    for(contador) in range(1,limite+1):

        resultado = tabla * contador

        msg += f'{tabla} x {contador} = {resultado} '
        msg+= validar(resultado)
    return msg


def validar(resultado):
    if(resultado%2==0):
        return "El resultado es  par\n"
    else:
        return"El resultado es impar\n"


#función lambda
tablas_lamb = lambda tabla,limite: [f"{tabla} x {contador} = {validar_lamb(tabla*contador)}" for contador in range(1,limite+1)]

validar_lamb = lambda resultado: "El resultado es par" if resultado%2==0 else "El resultado es impar"
